packages/components/pie-modal/src/defs.ts

```diff
@@ -1,28 +1,21 @@
+export const headingLevels = ['h1', 'h2', 'h3', 'h4', 'h5', 'h6'] as const;
+
 export interface ModalProps {
-    /**
-     * the heading of the modal .
-     */
     heading: string;
-    /**
-     * the rendered heading tag of the modal header.
-     * @default h2
-     */
-    headingLevel: 'h1' | 'h2' | 'h3' | 'h4' | 'h5' | 'h6';
-    /**
-     * If `true`, the modal will be opened.
-     * @default false
-     */
-    isOpen?: boolean;
+    headingLevel: typeof headingLevels[number];
+    isOpen: boolean;
 }
 
 /**
- * Modal heading levels/tags
+ * Event name for when the modal is closed.
+ *
+ * @constant
  */
-export const headingLevels: Array<ModalProps['headingLevel']> = ['h1', 'h2', 'h3', 'h4', 'h5', 'h6'];
+export const ON_MODAL_CLOSE_EVENT = 'pie-modal-close';
```

Comment:
Not for this PR but we should probably consider using the form events once we complete that ticket. If they work just as good we could remove these i think.

Reply:
Yup one for the future I reckon :)